### DISCUSSION TITLE
support setting values in dBa

### DISCUSSION THREAD

**Discussion by n1trux** [hastinbe/i3-volume#25] [Ideas]
it'd be nice if you could set the volume not only in %, but also in dBa. `pavucontrol` displays values in dBa, too, and pulseaudio supports it. One probably has to write a wrapper for Alsa, though.

**Comment by hastinbe** [comment]
I did consider supporting dB volume in the past, but the demand for it hasn't justified the work to do it (this is the first request for it). I decided that supporting a single volume type was easier (differences between PA and alsa-utils) and that % volume was the most user-friendly for the average user.

With that said, I'm still open to it and welcome any PRs, as I want support for as many use cases as possible. It's a simple idea that requires quite a bit of work to implement.

**Comment by hastinbe** [comment]
Been awhile, but I added experimental support for dB in d3671c5d18216f0a58e92ab359c0527949a54c29 (in the `pipewire-only` branch)

- Auto-detection of dB vs percentage in commands
- dB support in `set`, `up`, `down` commands
- `-U` option for dB/percentage display
- All output formats support dB
- Notifications show dB when requested
- Config file support for persistent dB display

Features:

- `volume set -6dB` - set volume in dB
- `volume up 3dB` - increase by 3 dB
- `volume -U db output` - display in dB
- `VOLUME_DISPLAY_UNIT=db` in config - persistent dB display
